dali/operators/util/normal_distribution_op.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/util/normal_distribution_op.h"
+#include "dali/core/static_switch.h"
+
+namespace dali {
+
+
+DALI_SCHEMA(NormalDistribution)
+                .DocStr(R"code(Creates a tensor that consists of data distributed normally.
+This operator can be ran in 3 modes, which determine the shape of the output tensor:
+1. Providing an input batch to this operator resolves in a batch of output tensors,
+where the shapes correspond to the shapes of tensors in the input batch.
+2. Providing a custom shape as an argument resolves in a batch of output tensors,
+where every output tensor has the same shape.
+3. Providing no input arguments resolves in a batch of output scalars,
+distributed normally (similar to `CoinFlip` operator).)code")
+                .NumInput(0, 1)
+                .NumOutput(detail::kNumOutputs)
+                .AddOptionalArg(detail::kMean, R"code(Mean value of the distribution)code",
+                                0.f, true)
+                .AddOptionalArg(detail::kStddev,
+                                R"code(Standard deviation of the distribution)code",
+                                1.f, true)
+                .AddOptionalArg(detail::kShape,
+                                R"code(Shape of single output tensor in a batch)code",
+                                detail::kShapeDefaultValue, true)
+                .AddOptionalArg(arg_names::kDtype, R"code(Data type for the output)code",
+                                DALI_FLOAT);
+
+DALI_REGISTER_OPERATOR(NormalDistribution, NormalDistributionCpu, CPU);
+
+
+#define NORM_TYPES (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float, double)  // NOLINT
```

Comment:
If we are supporting anything I'd consider supporting float16 as well

Reply:
added